collect_app/src/androidTest/java/org/odk/collect/android/support/pages/FillBlankFormPage.java

```diff
@@ -53,28 +50,31 @@ public BlankFormSearchPage searchInBar(String query) {
     }
 
     public FillBlankFormPage checkIsFormSubtextDisplayed() {
-        onView(withIndex(withId(R.id.form_subtitle2), 0)).check(matches(isDisplayed()));
-        return this;
+        return WaitFor.waitFor(() -> {
+            assertTextNotDisplayed(R.string.no_items_display_forms);
+            onView(withIndex(withId(R.id.form_subtitle2), 0)).check(matches(isDisplayed()));
+            return this;
+        });
     }
 
     public FillBlankFormPage checkMapIconDisplayedForForm(String formName) {
-        onData(allOf(is(instanceOf(Cursor.class)), CursorMatchers.withRowString(DatabaseFormColumns.DISPLAY_NAME, is(formName))))
-                .onChildView(withId(R.id.map_button))
-                .check(matches(isDisplayed()));
+        onView(withId(R.id.formList))
+                .perform(RecyclerViewActions.actionOnItem(hasDescendant(allOf(withId(R.id.form_title), withText(formName))), scrollTo()))
+                .check(matches(hasDescendant(allOf(withId(R.id.map_button), withEffectiveVisibility(ViewMatchers.Visibility.VISIBLE)))));
         return this;
     }
 
     public FillBlankFormPage checkMapIconNotDisplayedForForm(String formName) {
-        onData(allOf(is(instanceOf(Cursor.class)), CursorMatchers.withRowString(DatabaseFormColumns.DISPLAY_NAME, is(formName))))
-                .onChildView(withId(R.id.map_button))
-                .check(matches(not(isDisplayed())));
+        onView(withId(R.id.formList))
+                .perform(RecyclerViewActions.actionOnItem(hasDescendant(allOf(withId(R.id.form_title), withText(formName))), scrollTo()))
+                .check(matches(hasDescendant(allOf(withId(R.id.map_button), withEffectiveVisibility(ViewMatchers.Visibility.GONE)))));
         return this;
     }
 
     public FormMapPage clickOnMapIconForForm(String formName) {
-        onData(allOf(is(instanceOf(Cursor.class)), CursorMatchers.withRowString(DatabaseFormColumns.DISPLAY_NAME, is(formName))))
-                .onChildView(withId(R.id.map_button))
-                .perform(click());
+        onView(withId(R.id.formList))
+                .perform(RecyclerViewActions.actionOnItem(hasDescendant(allOf(withId(R.id.form_title), withText(formName))), clickOnViewChild(R.id.map_button)));
```

Comment:
Could we use content description instead of ID here?

Reply:
What's the benefit? And which id are you talking about `R.id.formList` or `R.id.form_title` or `R.id.map_button`?

Reply:
> And which id are you talking about R.id.formList or R.id.form_title or R.id.map_button?

Sorry that wasn't clear. I meant for the `map_button` (the content description should be something like "Open form map"). I think we should also be able to match the item through just the form name itself rather than needing to specify that the name is in the `form_title`.

> What's the benefit?

1. Using IDs increases coupling between the test and the actual implementation - we want to avoid as much of that as possible to make change as easy as possible later
2. Using content descriptions instead of IDs on views without text verifies that we've configured that view to work correctly with screen readers